### PR TITLE
Added lerp_angle built-in function

### LINKED ISSUE #30564: Add a built-in function for lerping angle
Similar to Unity's [Mathf.LerpAngle](https://docs.unity3d.com/ScriptReference/Mathf.LerpAngle.html).
This is my implementation in GDScript:

```
static func lerp_angle(from, to, weight):
    return from + _short_angle_dist(from, to) * weight

static func _short_angle_dist(from, to):
    var max_angle = PI * 2
    var difference = fmod(to - from, max_angle)
    return fmod(2 * difference, max_angle) - difference
```

### PR DESCRIPTION
**Pull request by Chaosus**
Implemented new built-in `lerp_angle` similar to Unity function https://docs.unity3d.com/ScriptReference/Mathf.LerpAngle.html (but accept and returns radians instead of angles).

Fix #30564

### CODE REVIEW

**Comment by Xrayez** [comment]
Works the same as in GDScript for my use case, thanks!

File: modules/gdscript/doc_classes/@GDScript.xml [hunk 1]
```diff
@@ -604,6 +604,29 @@
...
+			</return>
+			<argument index="0" name="from" type="float">
+			</argument>
+			<argument index="1" name="to" type="float">
+			</argument>
+			<argument index="2" name="weight" type="float">
+			</argument>
+			<description>
+				Linearly interpolates between two angles (in radians) by a normalized value.
+				Similar to [func lerp] but interpolate correctly when the angles wrapped around 360 degrees.
```
Comment (akien-mga) [inline modules/gdscript/doc_classes/@GDScript.xml]: `[method lerp]`.

Comment (Chaosus) [reply modules/gdscript/doc_classes/@GDScript.xml]: Fixed

Comment (akien-mga) [reply modules/gdscript/doc_classes/@GDScript.xml]: "when the angles **wrap** around".
Also not sure if it should mention "360 degrees" if the method is in radians. Maybe mention `[constant @GDScript.TAU]`?

Comment (Chaosus) [reply modules/gdscript/doc_classes/@GDScript.xml]: Ok )

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**